make/default_compiler_options

```diff
@@ -0,0 +1,19 @@
+# Set default compiler options.
+# Users should only need to set these three variables for use.
+# - CC: The compiler to use. Expecting g++ or clang++.
+# - O: Optimization level. Valid values are {0, 1, 2, 3}.
+# - AR: archiver (must specify for cross-compiling)
+##
+CC = clang++
+O = 3
+O_STANC = 0
+AR = ar
+
+# CPPFLAGS are used for both C and C++ compilation
```

Comment:
@seantalts, did we want to fix this properly so we're using the implicit make variables properly? If so, although using `CPPFLAGS` does work, it's actually meant to control the C preprocessor, not meant to be flags that are used for both C and C++.
https://www.gnu.org/software/make/manual/make.html#Implicit-Variables

If we're trying to be correct, I'd rather have this be `CXXFLAGS` and maybe duplicate that to `CFLAGS`.